app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -102,36 +102,47 @@ def powered_on_in_provider?(vm)
     end
 
     def vm_boot_from_cdrom(operation, name)
-      operation.get_provider_destination.vm_service.start(
-        :vm => {
-          :os     => {
-            :boot => {
-              :devices => [OvirtSDK4::BootDevice::CDROM]
-            }
-          },
-          :cdroms => [
-            {
-              :id => name
-            }
-          ]
-        }
-      )
+      operation.destination.with_provider_object(VERSION_HASH) do |vm_service|
+        vm_service.start(
+          :vm => {
+            :os     => {
+              :boot => {
+                :devices => [OvirtSDK4::BootDevice::CDROM]
+              }
+            },
+            :cdroms => [
+              {
+                :id => name
+              }
+            ]
+          }
+        )
+      end
     rescue OvirtSDK4::Error
       raise OvirtServices::VmNotReadyToBoot
     end
 
+    def detach_floppy(operation)
+      operation.destination.with_provider_object(VERSION_HASH) do |vm_service|
+        vm = vm_service.get
+        vm.update(:payloads => [])
```

Comment:
The `update` method is a method of the service, not a method of the object. That means that this should be like this:

```ruby
vm_service.update(:payloads => [])
```